source/dub/commandline.d

```diff
@@ -382,11 +384,23 @@ int runDubCommandLine(string[] args)
 
 	logDiagnostic("DUB version %s", getDUBVersion());
 
-	version(Windows){
-		// rdmd uses $TEMP to compute a temporary path. since cygwin substitutes backslashes
-		// with slashes, this causes OPTLINK to fail (it thinks path segments are options)
-		// we substitute the other way around here to fix this.
-		environment["TEMP"] = environment["TEMP"].replace("/", "\\");
+	{
+		// Guarantee that this environment variable is set
+		//  this is specifically needed because of the Windows fix that follows this statement.
+		// While it probably isn't needed for all targets, it does simplify things a bit.
+		if ("TEMP" !in environment)
```

Comment:
why did you put this outside the `version(Windows)` block? Spawned processes might behave differently if they suddenly have this environment variable set on POSIX

Reply:
Generally speaking, I always prefer consistent behavior between platforms, otherwise its too easy to introduce platform-specific behavior that will break when things change or during porting.

However I can see that in this case with ``TMP`` it could add additional complications, and should leave it for Windows only.